### DISCUSSION TITLE
A Number module/helper

### DISCUSSION THREAD

**Discussion by ngan** [ffaker/ffaker#503] [General]
I had preliminary discussion with @Volosh1n here: https://github.com/ffaker/ffaker/issues/498

After spending some time to try to migrate our (very large monolithic) codebase from `Faker` to `FFaker`, I'm re-raising the issue because I'm starting to think it's worth it.

Consider the following...

Throughout our codebase, we want to generate random numbers with specific lengths or formats. Keep in mind that because I want X digits, the leading digit cannot be a 0. This means, I can't simply do: `FFaker.numerify("###")` if I want a 3-digit number.

This seems fine to me, and was the suggestion I found everywhere:
```ruby
Faker::Number.number(digits: 3)
# vs
rand(100..999)
```

However, it gets (IMO) ugly fairly quickly:
```ruby
Faker::Number.number(digits: 16)
# I would go crazy maintaining this:
rand(1_000_000_000_000_000..9_999_999_999_999_999)
```

There is the option of doing this...
```ruby
FFaker.numerify("1###############")
# or
FFaker.numerify("#{rand(1..9)}###############")
```

But at that point, I feel like the intention of the code is completely lost. "Why do you have a single digit in the beginning?" "Why are you rand-ing 1 digit?" "How many #s is in there?" I just doesn't seem clear and simple.

The other useful helper in `Faker::Number` is:
```
Faker::Number.decimal(l_digits: 3, r_digits: 3) #=> 181.843
```

Again, leading zeros (and trailing ones) are important here, and we run in to the the same problem as above. But, again IMO, the intention and clarity of `Faker` is way better than any workaround you go with.

In closing, can we please add a Number module/helper into `FFaker`? 🙇

**Comment by Volosh1n** [comment]
![image](https://user-images.githubusercontent.com/16169679/130140547-0bea6d67-d50d-4ed5-bdba-0e5e3aa20ad5.png)
I think such method could do the trick (even deterministic). Maybe it could fit FFaker::Number module...
Also, Number module could be useful really, especially in cases, when user wants generate different types, like integer, float, etc.

**Comment by Volosh1n** [comment]
Please, checkout #505 
What do you think?

**Reply by ngan** [reply]
This is awesome! Thank you for doing the hard work @Volosh1n. Those helpers go a long way…

**Comment by ngan** [comment]
@Volosh1n 👋 Any estimate on when a new version with be released with this new module?

**Comment by Volosh1n** [comment]
@nan don't know. We don't have something like a release plan, just release if plenty of things were added. I think new version will be released soon.

**Reply by ngan** [reply]
@Volosh1n @marocchino can we get a new release? We’d like to start using the new Number module.

**Reply by Volosh1n** [reply]
@ngan idk when I'll be able to create a new one. Feel free to use ffaker's main branch in gemfile

**Reply by Volosh1n** [reply]
@ngan #508 will close it

**Reply by Volosh1n** [reply]
@ngan released
